website/docs/r/nsxt_edgegateway_l2_vpn_tunnel.html.markdown

```diff
@@ -0,0 +1,174 @@
+  }
+
+  pre_shared_key = "secret_passphrase"
+}
+
+resource "vcd_nsxt_edgegateway_l2_vpn_tunnel" "client-session" {
+  org = "datacloud"
+
+  # Note that this is a different edge gateway, as one edge gateway
+  # can function only in SERVER or CLIENT mode
+  edge_gateway_id = data.vcd_nsxt_edgegateway.client-testing.id
+
+  name        = "client-session"
+  description = "example description"
+
+  session_mode = "CLIENT"
+  enabled      = true
+
+  # must be sub-allocated on the Edge Gateway
+  local_endpoint_ip  = "101.22.30.3"
+  remote_endpoint_ip = "1.2.2.3"
+
+  stretched_network {
+    network_id = data.vcd_network_routed_v2.test_network_client.id
+    # CLIENT sessions need to define a tunnel ID for every stretched network
+    tunnel_id = 1
+  }
+
+  stretched_network {
+    network_id = data.vcd_network_routed_v2.test_network_client_other.id
+    tunnel_id  = 2
+  }
+
+  # Be aware that if there are changes in the `server-session`, the peer_code
+  # will be updated as well, so `terraform apply` needs to run twice
+  peer_code = vcd_nsxt_edgegateway_l2_vpn_tunnel.server-session.peer_code
+}
+```
+
+## Argument Reference
+
+The following arguments are supported:
+
+* `org` - (Optional) The name of organization to use, optional if defined at 
+  provider level. Useful when connected as sysadmin working across different organisations
+* `edge_gateway_id` - (Required) The ID of the edge gateway (NSX-T only). 
+  Can be looked up using [`vcd_nsxt_edgegateway`](/providers/vmware/vcd/latest/docs/data-sources/nsxt_edgegateway) data source
+* `name` - (Required) The name of the tunnel.
+* `description` - (Optional) The description of the tunnel.
+* `session_mode` - (Required) Mode of the tunnel session (SERVER or CLIENT)
+* `enabled` - (Optional) State of the `SERVER` session, always set to `true` for `CLIENT` 
+  sessions. Default is `true`.
+* `connector_initiation_mode` - (Required for `SERVER` sessions) Mode in which 
+  the connection is formed. Only relevant to `SERVER` sessions. One of:
+	* `INITIATOR` - Local endpoint initiates tunnel setup and will also respond to 
+  incoming tunnel setup requests from the peer gateway.
+	* `RESPOND_ONLY` - Local endpoint shall only respond to incoming tunnel setup 
+  requests, it shall not initiate the tunnel setup.
+	* `ON_DEMAND` - In this mode local endpoint will initiate tunnel creation once 
+  first packet matching the policy rule is received, and will also respond to 
+  incoming initiation requests.
+* `local_endpoint_ip` - (Required) The IP address corresponding to the Edge 
+  Gateway the tunnel is being configured on. The IP must be sub-allocated 
+  on the Edge Gateway.
+* `remote_endpoint_ip` - (Required) The IP address of the remote endpoint, which 
+corresponds to the device on the remote site terminating the VPN tunnel.
+* `tunnel_interface` - (Optional) The network CIDR block over which the session 
+  interfaces. Relevant only for SERVER session modes. If not provided, Cloud 
```

Comment:
I like "SERVER/CLIENT session mode" usage here, because it is clearer that we're referring to `session_mode` field in practice. All my comments in this review relate to this.

Reply:
Thank you for all the polishes. Approving the PR 👍